### PR TITLE
Update rust_swig to HEAD

### PR DESCRIPTION
**Pull request by giarc3**
rust_swig 0.4.0 gives issues with JniInvalidValue, but that's been fixed in the current git version. 

- Replaced chrono-include.rs with the new version provided in rust_swig/jni_tests/src
- `#[derive(Clone)]` had to be added to the foreigner_class of `UserId` and `GroupId` to allow vectors of them to be used
- Updated dependencies and removed ones we didn't need.

Tests pass in ironoxide-java with `cargo t` and `sbt test`.
Tests pass in ironoxide-scala after publishing java locally and given the `libironoxide-java.so` binary

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ironoxide-java"
-version = "0.3.0"
+version = "0.5.0"
```
Comment (clintfred) [inline Cargo.toml]: This is to sync the Rust and Scala versions to 0.5?

File: src/lib.rs.in [hunk 2]
```diff
@@ -41,6 +40,7 @@ class DeviceSigningKeyPair {
 
 foreigner_class!(
 /// ID of a user. Unique with in a segment. Must match the regex `^[a-zA-Z0-9_.$#|@/:;=+'-]+$`.
+#[derive(Clone)]
```
Comment (clintfred) [inline src/lib.rs.in]: Do you understand why this is needed now but not before?

Comment (giarc3) [reply src/lib.rs.in]: I don't. I was getting an error that it didn't know how to make a Vec<GroupId> or Vec<UserId>, so I looked at his example tests and noticed that his foreigner classes had this so I added it in and it fixed the issue.

File: Cargo.toml (hunk 1 above)
Comment (giarc3) [reply Cargo.toml]: Correct

File: src/lib.rs.in (hunk 2 above)
Comment (clintfred) [reply src/lib.rs.in]: This possibly means that the old version was handing back values that weren't safe for the JVM. 
Not sure how we can test this... Maybe something like: get a List back in Scala, grab a reference to one of the values and then empty the list (or set it to null, or similar) and make sure the element value still works properly.

Comment (giarc3) [reply src/lib.rs.in]: Here's the commit where he adds it:
https://github.com/Dushistov/rust_swig/commit/d29855ba0ef87923c186fe12aecc16d7e1329e2e

**Pull request merged**